internal/core/algorithm/ngt/option_test.go

```diff
@@ -373,27 +375,34 @@ func TestWithDimension(t *testing.T) {
 			},
 			want: want{
 				obj: &T{},
-				err: errors.New("invalid critical option, name: dimension, val: -100: dimension size -100 is invalid, the supporting dimension size must be between 2 ~ 65536"),
+				err: errors.New("invalid critical option, name: dimension, val: -100: dimension size -100 is invalid, the supporting dimension size must be between 2 ~ " + strconv.Itoa(maxDim)),
 			},
 		},
 		{
-			name: "return error when the size is 65537",
+			name: "return error when the size is larger than maxDim",
 			args: args{
-				size: 65537,
+				size: 1 << 32,
```

Comment:
```suggestion
				size: maxDim + 1,
```
Why don't you use `maxDim + 1`?